Userland/Libraries/LibImageDecoderClient/Client.cpp

```diff
@@ -74,13 +74,7 @@ void Client::did_decode_image(i64 image_id, bool is_animated, u32 loop_count, Ve
     image.scale = scale;
     image.frames.ensure_capacity(bitmaps.size());
     for (size_t i = 0; i < bitmaps.size(); ++i) {
-        if (!bitmaps[i].is_valid()) {
-            dbgln("ImageDecoderClient: Invalid bitmap for request {} at index {}", image_id, i);
-            promise->reject(Error::from_string_literal("Invalid bitmap"));
-            return;
-        }
-
-        image.frames.empend(*bitmaps[i].bitmap(), durations[i]);
+        image.frames.empend(*bitmaps[i], durations[i]);
```

Comment:
Do we need to check whether `bitmaps[i].has_value()` here or do we expect the optional to always have a value at this point?

Reply:
Oh, looks like we need to be checking it!

Reply:
I think we hit this when one of the images failed to decode. We end up throwing away all the valid ones in the process, though, so room for improvement